services/dataAccessLayer.js

```diff
@@ -0,0 +1,34 @@
+const userQuery = require("../models/users");
+
+const retrieveUsers = async ({ id = null, usernames = null, req = null }) => {
+  if (id) {
+    const result = await userQuery.fetchUser({ userId: id });
+    removeSensitiveInfo(result.user);
+    return result;
+  } else if (usernames) {
+    const { users } = await userQuery.fetchUsers(usernames);
+    users.forEach((element) => {
+      removeSensitiveInfo(element);
+    });
+    return users;
+  } else {
+    const { allUsers, nextId, prevId } = await userQuery.fetchPaginatedUsers(req.query);
+    allUsers.forEach((element) => {
+      removeSensitiveInfo(element);
+    });
+    return { allUsers, nextId, prevId };
+  }
+};
+
+const removeSensitiveInfo = function (obj, properties = ["phone", "email", "chaincode", "tokens"]) {
```

Comment:
properties[i] property exists in the obj object. However, the hasOwnProperty method only checks if the property is defined in the object's prototype chain. This means that the hasOwnProperty method will not return true if the property is defined directly in the obj object.

To fix this bug, you can use the in operator instead of the hasOwnProperty method. The in operator checks if the property exists in the object, regardless of whether it is defined in the object's prototype chain